File: python/inheritance/animal.py
class Animal:
        _name= ""
        _height= 0

        def __init__(self, name, height):
            self._name=name
            self._height=height
        
        def __str__(self):

            return "name:{} \nheight:{}\n".format(self._name, self._height)

        def get_name(self):
            return self._name
        def set_name(self, name):
            self._name = name

        def get_height(self):
            return self._name
        def set_height(self, height):
            self._height = height
        
        def get_type(self):
            print("Animal")

File: c++/inheritance/src/animal.cpp
#include "animal.h"

animal::animal(string name)
{
    this->name = name;
}

string animal::get_name(){
    return this->name;
}

void animal::set_name(string name){
    this->name = name;
}

string animal::toString()const{
    return string("name:"+this->name);
}

ostream& operator<<(ostream& out, const animal& obj){
	return out << obj.toString();
    // return out <<"name:"a.name;
}

animal::~animal()
{
}